src/web/rustdoc.rs

```diff
@@ -259,20 +261,46 @@ pub fn rustdoc_html_server_handler(req: &mut Request) -> IronResult<Response> {
 
     content.full = file_content;
     let crate_details = cexpect!(CrateDetails::new(&conn, &name, &version));
-    let latest_version = latest_version(&crate_details.versions, &version);
+    let (path, version) = if let Some(version) = latest_version(&crate_details.versions, &version) {
+        req_path[2] = &version;
+        (path_for_version(&req_path, &crate_details.target_name, &conn), version)
+    } else {
+        Default::default()
+    };
 
     content.crate_details = Some(crate_details);
 
     Page::new(content)
         .set_true("show_package_navigation")
         .set_true("package_navigation_documentation_tab")
         .set_true("package_navigation_show_platforms_tab")
-        .set_bool("is_latest_version", latest_version.is_none())
-        .set("latest_version", &latest_version.unwrap_or(String::new()))
+        .set_bool("is_latest_version", path.is_empty())
+        .set("path_in_latest", &path)
+        .set("latest_version", &version)
         .to_resp("rustdoc")
 }
 
-
+fn path_for_version(req_path: &[&str], target_name: &str, conn: &Connection) -> String {
```

Comment:
Can this function get a doc comment? Something like this would work:

```rust
/// Checks whether the given path (including version and target information) exists. Crate target
/// name is used to confirm whether a platform triple is part of the path.
///
/// Returns a path that can be appended to `/crate/version/` to create a complete URL.
```